posthog/hogql/expr_parser.py

```diff
@@ -0,0 +1,187 @@
+            response = (
+                f"divide({translate_ast(node.left, stack, context)}, {translate_ast(node.right, stack, context)})"
+            )
+        else:
+            response = f"({translate_ast(node.left, stack, context)} {translate_ast(node.op, stack, context)} {translate_ast(node.right, stack, context)})"
+    elif type(node) == ast.UnaryOp:
+        response = f"{translate_ast(node.op, stack, context)}{translate_ast(node.operand, stack, context)}"
+    elif type(node) == ast.USub:
+        response = "-"
+    elif type(node) == ast.Constant:
+        if type(node.value) == int or type(node.value) == float:
+            response = str(node.value)
+        elif type(node.value) == str or type(node.value) == list:
+            response = escape_param(node.value)
+        else:
+            raise ValueError(f"Unknown AST Constant node type '{type(node.value)}' for value '{str(node.value)}'")
+    elif type(node) == ast.Attribute or type(node) == ast.Subscript:
+        attribute_chain: list[str] = []
+        while True:
+            if type(node) == ast.Attribute:
+                attribute_chain.insert(0, node.attr)
+                node = node.value
+            elif type(node) == ast.Subscript:
+                slice: ast.AST = node.slice
+                if type(slice) == ast.Constant:
+                    if type(slice.value) != str:
+                        raise ValueError(f"Only string property access is currently supported, found '{slice.value}'")
+                    attribute_chain.insert(0, slice.value)
+                    node = node.value
+                # ast.Index is a deprecated node class that shows up in tests with Python 3.8
+                elif type(slice) == ast.Index and type(slice.value) == ast.Constant:  # type: ignore
+                    const: ast.Constant = slice.value  # type: ignore
+                    if type(const.value) != str:
+                        raise ValueError(f"Only string property access is currently supported, found '{const.value}'")
+                    attribute_chain.insert(0, const.value)
+                    node = const
+                else:
+                    raise ValueError(f"Unsupported Subscript slice type: {type(node.slice).__name__}")
+            elif type(node) == ast.Name:  # type: ignore
+                attribute_chain.insert(0, node.id)
+                break
+            else:
+                raise ValueError(f"Unknown node in field access chain: {ast.dump(node)}")
+        response = property_access_to_clickhouse(attribute_chain)
+        context.property_list.append(attribute_chain)
+
+    elif type(node) == ast.Call:
+        if type(node.func) != ast.Name:
+            raise ValueError(f"Can only call simple functions like 'avg(properties.bla)' or 'total()'")
+        call_name = node.func.id
+        if call_name in HOGQL_AGGREGATIONS:
+            context.is_aggregation = True
+            if call_name == "total":
```

Comment:
Why not call it just `count`?

Reply:
I think calling it `count` would be lovely. I left it as `total` now for parity with this block:

```ts
export enum BaseMathType {
    TotalCount = 'total', // implemented
    UniqueUsers = 'dau',
    WeeklyActiveUsers = 'weekly_active',
    MonthlyActiveUsers = 'monthly_active',
    UniqueSessions = 'unique_session',
}

export enum PropertyMathType {
    Average = 'avg', // implemented
    Sum = 'sum', // implemented
    Minimum = 'min',
    Maximum = 'max',
    Median = 'median',
    P90 = 'p90',
    P95 = 'p95',
    P99 = 'p99',
}

export enum CountPerActorMathType {
    Average = 'avg_count_per_actor',
    Minimum = 'min_count_per_actor',
    Maximum = 'max_count_per_actor',
    Median = 'median_count_per_actor',
    P90 = 'p90_count_per_actor',
    P95 = 'p95_count_per_actor',
    P99 = 'p99_count_per_actor',
}
```

I hate it when standard things are a bit different, such as `null` vs `None` vs `nil`, so I would be in favor of calling it `count`... yet it's also a bit different since we won't support `count(bla)` or `count(distinct field)`, at least not soon... 🤔

I will happily implement all the other non-implemented aggregations and functions in a follow up PR. As for changing `total()` to `count()`... 🤷. Why not, but let's have buy in from everyone and make sure it's in a system that makes sense...